tests/functional/interfaces-with-arguments.ts

```diff
@@ -0,0 +1,105 @@
+import "reflect-metadata";
+import { getMetadataStorage } from "../../src/metadata/getMetadataStorage";
+import {
+  IntrospectionSchema,
+  IntrospectionObjectType,
+  IntrospectionInterfaceType,
+  IntrospectionField,
+} from "graphql";
+import { getSchemaInfo } from "../helpers/getSchemaInfo";
+import { ArgsType, Field, Int, InterfaceType, ID, Query, ObjectType, Arg, Args } from "../../src";
+
+describe("Interfaces with arguments", () => {
+  describe("Schema", () => {
+    let schemaIntrospection: IntrospectionSchema;
+    let queryType: IntrospectionObjectType;
+
+    beforeAll(async () => {
+      getMetadataStorage().clear();
+
+      @ArgsType()
+      class ImageArgs {
+        @Field(type => Int, { nullable: true })
+        width?: number;
+        @Field(type => Int, { nullable: true })
+        height?: number;
+      }
+
+      @InterfaceType()
+      abstract class Article {
+        @Field(type => ID)
+        id: string;
+        @Field({ nullable: true })
+        headline?: string;
+
+        @Field()
+        image1(@Arg("width") width: number, @Arg("height") height: number): string {
+          return `http://lorempixel.com/${width}/${height}/`;
+        }
+        @Field()
+        image2(@Args(type => ImageArgs) args: any): string {
+          return `http://lorempixel.com/${args.width}/${args.height}/`;
+        }
+      }
+
+      @ObjectType({ implements: Article })
+      class WebArticle extends Article {
+        @Field()
+        url: string;
+
+        @Field()
+        image3(@Arg("width") width: number, @Arg("height") height: number): string {
+          return `http://lorempixel.com/${width}/${height}/`;
+        }
+      }
+
+      class SampleResolver {
+        @Query(returns => [Article])
+        sampleQuery(): Article[] {
+          return [];
```

Comment:
You should test the resolvers too, not only the signature - does it support resolvers inheritance? Will it call the method from super class correctly?

Reply:
you are totally right - and there is an issue with resolving fields of interfaces when using inline arguments. I will have to look deeper into that another day. But I already committed and pushed functional tests for the resolver.

Reply:
this fixed - it was missing the field resolvers - I should have used the same arguments as for the object type from the start ;)